packages/app-mobile/components/screens/Notes.tsx

```diff
@@ -197,6 +199,29 @@ class NotesScreenComponent extends BaseScreenComponent<any> {
 		return this.folderPickerOptions_;
 	}
 
+	public menuOptions() {
+		const output: MenuOptionType[] = [];
+
+		if (this.props.notesParentType === 'Tag') {
+			output.push({
+				title: _('Copy external link'),
+				onPress: () => {
+					Clipboard.setString(getTagCallbackUrl(this.props.selectedTagId));
+				},
+			});
+		}
+		if (this.props.notesParentType === 'Folder') {
+			output.push({
+				title: _('Copy external link'),
+				onPress: () => {
+					Clipboard.setString(getFolderCallbackUrl(this.props.selectedFolderId));
+				},
+			});
+		}
```

Comment:
For consistency with the existing UI, this menu option should appear when long pressing a folder in the sidebar.

Same for the tags, although currently we don't have a menu there.

Reply:
When using the mobile app, I never thought to long press on the item in the sidebar, or to long press the tag in the list when I want to look for the "Copy external link" function.

I used "Copy markdown link" function before, it's in the right-top menu button, so I looked there and add the function there. Folder/tag node list didn't have that menu button before but I think it's natural to look for the function at the same place. Mental model ls like "copy the link to current screen".

Reply:
That makes sense, but the problem is that we'd now have similar functionalities in two different places. Either we move all the long press features to the kebab menu, or we move that copy external link item to the long press one.

Reply:
> this menu option should appear when long pressing a folder in the sidebar.
I need clarification on this, since in the mobile app what I found while long pressing a folder in the sidebar is the alert dialog I took screenshot above.

Tags doesn't show in the sidebar, but in a tag list on the right side (I'm not sure what that part called). Long-pressing a tag currently shows nothing.

Long-pressing a note will change the title bar to 'Move to notebook ...' dropdown, select all button, delete button, copy button; there's no space for additional buttons or existing menu to add new action to.

I can't think a way to put 'Copy external link' into these places that feels 'consistent'. What should it 'consistent' with?

Reply:
I don't have an answer to all these questions unfortunately. It's possible there is work to do on mobile app to allow some of this. It's not nicely setup so that we can easily add an option to a list.

The fact that there are notebook-related actions in the long press menu, and that you want to add more notebook-related actions in a separate place I think is obviously inconsistent? This is what I'm talking about. No idea about tags, etc.

Reply:
What I feel is that 'Copy external link' in the kebab menu is consistent (that's what I first go looking when I want to copy a link for notebook).

Perhaps convert the long-press alert dialog of the notebook in the sidebar to a dropdown menu with Edit and Delete option like the kebab menu and desktop version's right-click menu?

Or I just remove the tag and notebook options from this pull request, since I can and would produce them from the desktop version anyway, it's just a convenience function in the mobile app since there's label making app on mobile and I do have a few friends would produce labels directly on a phone.

The core feature I want to have is callback URL handling, we can continue the "where to put the new menu option" discussion later if needed ...

Reply:
Joplin callback URL works fine on Windows, but doesn't work on Android phone (Chrome doesn't recognize joplin:// handler). My use case that will enable me to migrate from Evernote is to be able to link my Android Contacts URLs with respective contact note. Currently it works with Evernote public https URL, works with joplin:// handler in Thunderbird contacts, but when synced to my google account contacts, handler jopin:// is unrecognized by chrome nor firefox mobile. Any chance to suggestion how to successfully use joplin:// in Android?

Reply:
So we're trying to improve the mobile app UX and adding this menu there, as a duplicate of the long press menu, seems counter productive to me. If we do that, we'll have to fix it anyway when we go ahead and refactor the mobile app.

> Perhaps convert the long-press alert dialog of the notebook in the sidebar to a dropdown menu with Edit and Delete option like the kebab menu and desktop version's right-click menu?

I feel that would be a good solution! Perhaps another option is to use something, a dialog or menu from the react-native-paper library that we now use? And if so, @tiberiusteng would you be ok with making this change? There's also the menu that popup when pressing on the filter icon in the top right corner of the note list. I'm not sure how this one is done but it looks like it can have many options.

Reply:
Actually I've tried to implement that but my knowledge/ability on React-Native isn't good enough to finish it (yet) ... so it probably need to be implemented later by somebody else.

Personally I still think putting the "Copy external link" option on the top-right dotted menu is the most intuitive place, because they appear at the same place for note, (note list of) tag, (note list of) notebook, and serve the idea of "Copy link to current screen", instead of scattering across:

* For note, top-right dotted menu
* For notebook, long-press notebook name in sidebar menu (it looks like menu in iOS but a strange dialog in Android, should be fixed anyway)
* For tag, long-press tag name in tag list (which don't do anything right now)

Anyway the core part is handling external links on mobile, the menus are nice to have but perhaps I've done a little too much. If it's not desired I can add a new commit to remove them, or just don't merge `Note.tsx` and `Notes.tsx`.

Reply:
Ok I think it's reasonable too to have this context menu as you suggest. But in that case would you be able to remove the long-press menu and move the options over to that kebab menu?

Reply:
OK I'll try it.

Reply:
Thanks @tiberiusteng. It looks like you made further changes and added the kebab menu, but the long press menu is still there? If could remove it now that you moved the functionalities that would great.

Reply:
I was thinking that removing existing interface will cause confusion to users used that function before (but I don't think of a good way to direct users from original position to the new position, maybe just let them co-exist a few versions). But if you are OK to remove it now it's great too. I will wait for further discussion or final confirmation.